pkg/controller/build/reconciler.go

```diff
@@ -791,6 +911,150 @@ func (b *buildReconciler) deleteMachineOSBuild(ctx context.Context, mosb *mcfgv1
 	return fmt.Errorf("could not delete MachineOSBuild %s for MachineOSConfig %s: %w", mosb.Name, moscName, err)
 }
 
+func (b *buildReconciler) getAuthFilePath(mosb *mcfgv1.MachineOSBuild, moscName string) (string, error) {
+	pushSecret := mosb.GetAnnotations()[constants.RenderedImagePushSecretAnnotationKey]
+	secret, err := b.kubeclient.CoreV1().Secrets(ctrlcommon.MCONamespace).Get(context.TODO(), pushSecret, metav1.GetOptions{})
+	if err != nil {
+		return "", fmt.Errorf("could not get rendered push secret for MachineOSConfig %q: %w", moscName, err)
+	}
+	if secret.Type != corev1.SecretTypeDockerConfigJson {
+		return "", fmt.Errorf("rendered push secret for MachineOSConfig %q is not of type %q, cannot delete image", moscName, corev1.SecretTypeDockerConfigJson)
+	}
+	var authConfig map[string]interface{}
+	data, ok := secret.Data[corev1.DockerConfigJsonKey]
+	if !ok {
+		return "", fmt.Errorf("rendered push secret for MachineOSConfig %q does not have key %q, cannot delete image", moscName, corev1.DockerConfigJsonKey)
+	}
+	if err := json.Unmarshal(data, &authConfig); err != nil {
+		return "", fmt.Errorf("could not unmarshal rendered push secret for MachineOSConfig %q: %w, cannot delete image", moscName, err)
+	}
+	// Create a temp auth.json file
```

Comment:
thought (non-blocking): I've run into this pattern before. I really wish `containers/image` had a better way to do this.